### PR TITLE
refactor: Switching to Expo

### PR DESCRIPTION
**Pull request by Koalk**
Facilitate the increments in native libraries and use prebuild for the example app to reduce the amount of managed code in this repo.

General refactor of the library to use expo and expo prebuild.
Closes https://github.com/PlayerData/react-native-mcu-manager/issues/70

---------------------

Self Review:

* [x] Appropriate test coverage
* [x] Relevant Documentation updated

Smoke Tests:

* [x] Tested with the example app
* [x] Tested as an integrated library on another app

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,80 +1,36 @@
...
 
-## Getting started
+# API documentation
 
-`$ npm install @playerdata/react-native-mcu-manager --save`
-## Usage
-```ts
-import McuManager, { ProgressEvent, UploadEvents } from '@playerdata/react-native-mcu-manager';
+- [Documentation for the main branch](https://github.com/expo/expo/blob/main/docs/pages/versions/unversioned/sdk/@playerdata/react-native-mcu-manager.md)
+- [Documentation for the latest stable release](https://docs.expo.dev/versions/latest/sdk/@playerdata/react-native-mcu-manager/)
```
Comment (ball-hayden) [inline README.md:8]: We should remove these links.

The old `Usage` section is probably still useful.

File: example/App.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
+import { UpgradeMode } from '@playerdata/react-native-mcu-manager';
+import * as Location from 'expo-location';
```
Comment (ball-hayden) [inline example/App.tsx:2]: Do we need location?

File: example/src/BLEService.ts [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+            resolve();
+            subscription.remove();
+            break;
+          default:
+            console.error('Unsupported state: ', state);
+        }
+      }, true);
+    });
+
+  disconnectDevice = () => {
```
Comment (ball-hayden) [inline example/src/BLEService.ts:66]: We should refactor this to be an `async` function.

File: example/src/BLEService.ts [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+      .cancelDeviceConnection(this.device.id)
+      .then(() => this.showSuccessToast('Device disconnected'))
+      .catch((error) => {
+        if (error?.code !== BleErrorCode.DeviceDisconnected) {
+          this.onError(error);
+        }
+      });
+  };
+
+  disconnectDeviceById = (id: DeviceId) =>
```
Comment (ball-hayden) [inline example/src/BLEService.ts:81]: This can also be `async`?

File: example/src/BLEService.ts [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+            resolve(this.device);
+          } else {
+            this.onError(error);
+            reject(error);
+          }
+        });
+    });
+
+  discoverAllServicesAndCharacteristicsForDevice = async () =>
+    new Promise<Device>((resolve, reject) => {
```
Comment (ball-hayden) [inline example/src/BLEService.ts:136]: We can simplify this with `await` and avoid returning the promise.

Not entirely sure what the behaviour of an async function returning a promise is, but suspect it's a double promise.

File: example/ios/Podfile [hunk 6]
```diff
@@ -1,34 +1,87 @@
-require_relative '../node_modules/react-native/scripts/react_native_pods'
-require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
+require File.join(File.dirname(`node --print "require.resolve('expo/package.json')"`), "scripts/autolinking")
```
Comment (ball-hayden) [inline example/ios/Podfile:1]: Shouldn't this be gitignored now?

**Review by saty9 — DISMISSED** [review]

**Review by ball-hayden — CHANGES_REQUESTED** [review]
We've ended up with bits of the example app back in.

Also the commit message needs rewording - `BREAKING CHANGE` needs to be in the footer (i.e. following at least one newline) e.g.:

```
feat(expo): Switch to an Expo Module structure

BREAKING CHANGE: Requires that the using app is set up to use Expo Modules
```

File: example/ios/Podfile (hunk 6 above)
Comment (Koalk) [reply example/ios/Podfile:1]: yep... eveything inside example/ios and example/android should be deleted and ignored now.

**Comment by Koalk** [comment]
> We've ended up with bits of the example app back in.
> 
> Also the commit message needs rewording - `BREAKING CHANGE` needs to be in the footer (i.e. following at least one newline) e.g.:
> 
> ```
> feat(expo): Switch to an Expo Module structure
> 
> BREAKING CHANGE: Requires that the using app is set up to use Expo Modules
> ```

commit lint didn't let me make a commit message that was that big. I'll do it with the no verify flag if I can't get it in otherwise

**Review by ball-hayden — APPROVED** [review]

**Pull request merged**